### PR TITLE
Marcos.cifuentes/setting splash screen

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
In this PR I added the Splash Screen for the app

- Updated Ktlint to the latest version
- Created the layout and activity for the Splash Screen
- Added Glide, coroutines and Lottie dependencies to handle the resources of the Splash Screen
- Implemented MVVM architecture (ViewModel)
- Added Koin to do injections
- Added unit test

Test passing and coverage:

test passed

test coverage

Splash Screen GIF:

pokemon splash screen

Smaller device:

pokemon splash screen smaller device

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[*.{kt,kts}]
+disabled_rules=import-ordering
```
Comment (lucianoeli) [inline .editorconfig:2]: why??  :(

Comment (MarcosCGlobant) [reply .editorconfig:2]: that line removes the alphabetical imports ordering in ktlint

File: app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = ActivitySplashScreenLayoutBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        activityScope.launch {
+
+            Glide.with(applicationContext).load(R.drawable.app_logo).into(binding.activitySplashScreenTitleImageView)
+
+            delay(3000)
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt]: add this as a const val pleease

Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt]: added, thanks

File: app/build.gradle [hunk 3]
```diff
@@ -6,6 +6,10 @@ android {
     compileSdkVersion 29
     buildToolsVersion "29.0.3"
 
+    buildFeatures {
+        viewBinding = true
+    }
+
     defaultConfig {
         applicationId "com.example.pokemontcgapp"
```
Comment (SartiMau) [inline app/build.gradle]: please use the package as com.globant. pokemontcgapp

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -4,18 +4,22 @@
...
-        android:roundIcon="@mipmap/ic_launcher_round"
+        android:largeHeap="true"
+        android:roundIcon="@drawable/app_logo"
         android:supportsRtl="true"
-        android:theme="@style/AppTheme">
-        <activity android:name="com.globant.pokemontcgapp.MainActivity">
+        android:theme="@style/AppTheme.NoActionBar">
+        <activity
+            android:name="com.globant.pokemontcgapp.activity.SplashScreenActivity"
+            android:hardwareAccelerated="true">
```
Comment (SartiMau) [inline app/src/main/AndroidManifest.xml]: why do you need this line?

**Review by lucianoeli — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

File: app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.bumptech.glide.Glide
+import com.globant.pokemontcgapp.R
+import com.globant.pokemontcgapp.databinding.ActivitySplashScreenLayoutBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class SplashScreenActivity : AppCompatActivity() {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt:14]: you must implement MVVM

**Review by SartiMau — CHANGES_REQUESTED** [review]
Review comments

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (MarcosCGlobant) [reply app/src/main/AndroidManifest.xml]: that was an old fix i used to help the animations run smoother, with Lottie i don't need it, thanks for the catch

File: app/build.gradle (hunk 3 above)
Comment (MarcosCGlobant) [reply app/build.gradle]: changed, thanks

File: app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt (hunk 5 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/activity/SplashScreenActivity.kt:14]: implemented, thanks

File: app/src/main/res/drawable/gradient_background.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <gradient
+        android:angle="270"
+        android:endColor="#65aaeb"
```
Comment (mauroLonghi) [inline app/src/main/res/drawable/gradient_background.xml]: You should add a resource value

File: app/src/main/java/com/globant/pokemontcgapp/activity/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.globant.pokemontcgapp.R
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+    }
+
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/pokemontcgapp/activity/MainActivity.kt:16]: Pls add a //TODO

File: app/src/main/res/drawable/gradient_background.xml (hunk 6 above)
Comment (MarcosCGlobant) [reply app/src/main/res/drawable/gradient_background.xml]: added, thanks

**Review by SartiMau — COMMENTED** [review]
Please, add the unit tests to the SplashScreenViewModel

File: app/src/main/java/com/globant/pokemontcgapp/activity/MainActivity.kt (hunk 7 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/activity/MainActivity.kt:16]: added, thanks

File: app/src/main/java/com/globant/pokemontcgapp/viewmodel/SplashScreenViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import androidx.lifecycle.viewModelScope
+import com.globant.pokemontcgapp.contract.SplashScreenContract
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class SplashScreenViewModel : ViewModel(), SplashScreenContract.ViewModel {
+
+    private var mutableMainState: MutableLiveData<SplashScreenData> = MutableLiveData()
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/pokemontcgapp/viewmodel/SplashScreenViewModel.kt]: i think this can hold a SplashScreenStatus and avoid th SPlashScreenData

Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/viewmodel/SplashScreenViewModel.kt]: changed, thanks

File: app/src/test/java/com/globant/pokemontcgapp/splashscreentest/SplashScreenViewModelTest.kt [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    @ObsoleteCoroutinesApi
+    private var mainThreadSurrogate = newSingleThreadContext(UI_THREAD)
+
+    @get:Rule
+    val taskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var viewModel: SplashScreenContract.ViewModel
+
+    @ObsoleteCoroutinesApi
+    @ExperimentalCoroutinesApi
```
Comment (lucianoeli) [inline app/src/test/java/com/globant/pokemontcgapp/splashscreentest/SplashScreenViewModelTest.kt]: why is this needed?

Comment (MarcosCGlobant) [reply app/src/test/java/com/globant/pokemontcgapp/splashscreentest/SplashScreenViewModelTest.kt]: to remove warnings on the class

**Review by SartiMau — COMMENTED** [review]
Approved after Braian's comment

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]
